clang/lib/Format/Format.cpp

```diff
@@ -3957,7 +3958,12 @@ LangOptions getFormattingLangOpts(const FormatStyle &Style) {
   LangOpts.Digraphs = LexingStd >= FormatStyle::LS_Cpp11;
 
   LangOpts.LineComment = 1;
-  LangOpts.CXXOperatorNames = Style.isCpp();
+
+  const auto Language = Style.Language;
+  LangOpts.C17 = Language == FormatStyle::LK_C;
```

Comment:
Shouldn't `LangOpts.CPlusPlus` be false for C inputs? But I imagine that could require some additional changes

Reply:
Exactly. See #129426.